kolibri/core/public/utils.py

```diff
@@ -171,35 +211,47 @@ def request_soud_sync(server, user=None, queue_id=None, ttl=10):
+        return  # Request done to a server not owning this user's data
+
+    if response.status_code == status.HTTP_200_OK:
+        handle_server_sync_response(response, server, user)
+
+
+def handle_server_sync_response(response, server, user):
+    # In either case, we set the sync status for this user as queued
+    # Once the sync starts, then this should get cleared and the SyncSession
+    # set on the status, so that more info can be garnered.
     response_content = (
         response.content.decode()
         if isinstance(response.content, bytes)
         else response.content
     )
     server_response = json.loads(response_content or "{}")
-    if response.status_code == status.HTTP_404_NOT_FOUND:
-        return  # Request done to a server not owning this user's data
+    UserSyncStatus.objects.update_or_create(user_id=user, defaults={"queued": True})
 
-    if response.status_code == status.HTTP_200_OK:
-        # In either case, we set the sync status for this user as queued
-        # Once the sync starts, then this should get cleared and the SyncSession
-        # set on the status, so that more info can be garnered.
-        UserSyncStatus.objects.update_or_create(user_id=user, defaults={"queued": True})
-
-        if server_response["action"] == SYNC:
-            job_id = startpeerusersync(server, user)
-            logger.info(
-                "Enqueuing a sync task for user {} in job {}".format(user, job_id)
+    if server_response["action"] == SYNC:
+        server_sync_interval = server_response.get(
+            "sync_interval", str(OPTIONS["Deployment"]["SYNC_INTERVAL"])
+        )
+        job_id = startpeerusersync(server, user, server_sync_interval)
+        logger.info("Enqueuing a sync task for user {} in job {}".format(user, job_id))
+
+    elif server_response["action"] == QUEUED:
+        pk = server_response["id"]
+        time_alive = server_response["keep_alive"]
+        dt = datetime.timedelta(seconds=int(time_alive))
+        job = Job(request_soud_sync, server, user, pk)
+        scheduler.enqueue_in(dt, job)
+        logger.info(
+            "Server busy, will try again in {} seconds with pk={}".format(
+                time_alive, pk
             )
+        )
 
-        elif server_response["action"] == QUEUED:
-            pk = server_response["id"]
-            time_alive = server_response["keep_alive"]
-            dt = datetime.timedelta(seconds=int(time_alive))
-            job = Job(request_soud_sync, server, user, pk)
-            scheduler.enqueue_in(dt, job)
-            logger.info(
-                "Server busy, will try again in {} seconds with pk={}".format(
-                    time_alive, pk
-                )
-            )
+
+def schedule_new_sync(server, user, interval=OPTIONS["Deployment"]["SYNC_INTERVAL"]):
+    # reschedule the process for a new sync
+    dt = datetime.timedelta(seconds=interval)
+    job = Job(request_soud_sync, server, user)
```

Comment:
One thing I realized as I was stopping and restarting my server for testing is that while we are deduplicating the actual sync jobs above, the scheduled checks are getting duplicated still here.

My scheduled jobs table is growing with each restart. Maybe an id of a hash of `server:user` might help to deduplicate?

Reply:
Done. The number of jobs now is reasonable. One thing I noticed is that restarting the server makes zeroconf pick a different url between all the ips available in the server machine, so with the same server sometimes it creates new hashes. 

I can think of cases where one learner syncs with different servers of different coaches in the same network  so it seems the server must be in the hash.